readthedocs/api/v3/views.py

```diff
@@ -384,30 +384,81 @@ def create(self, request, **kwargs):  # pylint: disable=arguments-differ
         return Response(data=data, status=code)
 
 
-class NotificationsViewSet(
+class NotificationsForUserViewSet(
+    APIv3Settings,
+    FlexFieldsMixin,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateMixin,
+    UpdateModelMixin,
+    GenericViewSet,
+):
+
+    """
+    Endpoint to return all the notifications related to the logged in user.
+
+    Hitting this endpoint while logged in will return notifications attached to:
+
+     - User making the request
+     - Organizations where the user is owner/member
+     - Projects where the user is admin/member
+    """
+
+    model = Notification
+    serializer_class = NotificationSerializer
+    queryset = Notification.objects.all()
+    permission_classes = (IsAuthenticated,)
+    filterset_class = NotificationFilter
+
+    def get_queryset(self):
+        return Notification.objects.for_user(self.request.user)
+
+
+# TODO: verify permissions over all ``**/notifications/`` endpoints.
```

Comment:
Do we have some test coverage here now? Maybe we might put this up as a follow up issue if it's still outstanding.

Reply:
I started working on more testing and better permissions today, but I haven't finished it yet.

I will have updates next week.

Reply:
I just pushed the permissions refactor and test cases for them. They follow our current pattern.